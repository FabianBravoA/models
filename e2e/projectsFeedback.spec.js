const mongoose = require('mongoose');
const ProjectFeedback = require('../src/models/ProjectFeedback');


describe('e2e::projectsFeedback', () => {
  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__);
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  it('should fail when missing props', () => {
    const projectFeedback = new ProjectFeedback({
      projectid: 'cipher',
      cohortid: 'lim-2019-01-bc-js-09',
      uid: 'xxxxxxxxxxxxxxxxxxxx',
      reviewerSurvey: 'xxxx',
      createdBy: 'lupo',
    });

    return projectFeedback.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it.only('should ...', () => {
    const projectJson = {
      projectid: 'cipher',
      cohortid: 'lim-2019-01-bc-js-09',
      uid: 'xxxxxxxxxxxxxxxxxxxx',
      reviewerSurvey: 'xxxx',
      createdBy: 'lupo',
      rubric: '2',
      rubricResults: {
        completion: 3,
        selfLearning: 2,
      },
    };
    const projectFeedback = new ProjectFeedback(projectJson);

    return projectFeedback.save()
      .then((result) => {
        const updatedProps = {
          reviewerSurveyResults: { perception: 'foo...' },
        };
        const { _id, createdAt, ...obj } = result.toJSON();
        expect(_id.constructor.name).toBe('ObjectID');
        expect(result.createdAt instanceof Date).toBe(true);
        expect(obj).toMatchSnapshot();
        // projectFeedback.reviewerSurveyResults = { perception: 'foo...' };
        // return projectFeedback.save();
        // return ProjectFeedback.findByIdAndUpdate(result._id, updatedProps);
        return projectFeedback.update(updatedProps);
      })
      .then((updateResult) => {
        expect(updateResult).toMatchSnapshot();
        return ProjectFeedback.find({
          projectid: projectJson.projectid,
          cohortid: projectJson.cohortid,
        }).exec();
      })
      .then((docs) => {
        expect(Array.isArray(docs)).toBe(true);
        expect(docs.length).toBe(1);
        expect(`${docs[0].toJSON()._id}`).toBe(`${projectFeedback.toJSON()._id}`);
      });
  });
});
