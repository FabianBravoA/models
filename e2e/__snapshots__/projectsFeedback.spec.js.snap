// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e::projectsFeedback should ... 1`] = `
Object {
  "__v": 0,
  "cohortid": "lim-2019-01-bc-js-09",
  "createdBy": "lupo",
  "projectid": "cipher",
  "reviewerSurvey": "xxxx",
  "rubric": "2",
  "rubricResults": Object {
    "completion": 3,
    "selfLearning": 2,
  },
  "uid": "xxxxxxxxxxxxxxxxxxxx",
}
`;

exports[`e2e::projectsFeedback should ... 2`] = `
Object {
  "n": 1,
  "nModified": 1,
  "ok": 1,
}
`;

exports[`e2e::projectsFeedback should fail when missing props 1`] = `
Object {
  "rubric": [ValidatorError: Path \`rubric\` is required.],
  "rubricResults": [ValidatorError: Path \`rubricResults\` is required.],
}
`;
